src/renderer/services/neo.js

```diff
@@ -573,6 +704,81 @@ export default {
           pushSystemAssetHolding(GAS_ASSET_ID, 'GAS', 0);
         }
 
+        const pushNewHolding = (assetId, name, symbol, balance, decimals,
+          canPull, isETHToken, abi, availableSupply) => {
+          const systemAssetHolding = {
+            assetId,
+            balance: new BigNumber(balance),
+            symbol,
+            name,
+            contractBalance: new BigNumber(0),
+            totalBalance: new BigNumber(0),
+            decimals,
+            isUserAsset: true,
+            canPull,
+            isETHToken,
+            abi,
+            availableSupply,
+          };
+          holdings.push(systemAssetHolding);
+        };
+
+        // ETH System Holding
+
+        const ethAddress = currentWallet.ethAddress;
+        const balEthTemp = await web3.eth.getBalance(ethAddress);
+        const balEth = toBigNumber(balEthTemp).dividedBy(10 ** 18);
+
+        pushNewHolding('0x0000000000000000000000000000000000000000',
+          'Ethereum', 'ETH', balEth, 18, undefined, undefined, undefined, undefined);
+
+        // BTC System Holding
+
+        let balBTCtemp;
+        let balBTC;
+        const walletClient = currentWallet.btcWalletClient;
+
+        try {
+          balBTCtemp = await bwclient.getBTCBalanceByAddress(walletClient);
+          balBTC = balBTCtemp.availableAmount;
+        } catch (e) {
+          alerts.networkException(e);
```

Comment:
if this happens without returning and balBTCtemp stays as undefined it will cause NPE.

Reply:
no, it will not, catch will handle it, if it is undefined

Reply:
How so? The catch block does not return. It will continue past the catch block and below it tries to use balBTCtemp which will not have had a value set.